### PR TITLE
Skeleton for a TimeFrame class

### PR DESCRIPTION
**Pull request by sawenzel**
This is a working piece of deployable unit that brings in a skeleton TimeFrame class (including working ROOT IO + unit test).

This can now be used by the DataFlow effort and expanded thereafter.

### CODE REVIEW

**Comment by sawenzel** [comment]
The compile errors come from code which is not even in my pull request.

**Comment by ktf** [comment]
This is probably a mistake in FairRoot. We should not  `#include <zmq.h>` but more `#include "zmq.h"`.

File: DataFormats/Headers/include/Headers/TimeStamp.h [hunk 1]
```diff
@@ -102,7 +102,7 @@ class TimeStamp
   typedef AliceO2::Header::Descriptor<2> TimeUnitID;
   // TODO: typedefs for the types of ticks and subticks
 
-  TimeStamp();
+  TimeStamp() {}
```
Comment (ktf) [inline DataFormats/Headers/include/Headers/TimeStamp.h]: = default;

Comment (sawenzel) [inline DataFormats/Headers/include/Headers/TimeStamp.h]: done

**Comment by Barthelemy** [comment]
> This is probably a mistake in FairRoot. We should not #include <zmq.h> but more #include "zmq.h".

I thought that we should use apostrophes for our own include files. When using apostrophes the current directory is searched before the default include directories and the one specified on the command line. I don't see why we need this for zmq.h.

**Comment by ktf** [comment]
I was wondering if this is actually the case also when zeromq is not found on the system but built by alibuild. However, in this particular situation, I think you are right and the error was simply that the `fairroot_base_bucket` does not depend on zeromq.

**Comment by mkrzewic** [comment]
Hi, about the TimeFrame class: I imagine this is for timeframe building, right? In any case: the incoming messages are owned by the transport, and should stay owned/managed by it since we probably want to forward those things to other devices. In that respect I think we should not keep a pointer to the buffer in MessageSizePair, but a (unique?)pointer to the message, which already knows it's size. We then end up with a std::vector<unique_ptr<FairMQMessage>> - which is exactly FairMQParts.
Now FairMQParts might be disappearing since fairmq want to get rid of multipart, so this particular typedef would go to O2, but the ownership semantics should be like this: We always let the transport allocate nad manage the memory for data exchange (since at compile time we don't know where the allocation will take place, shmem or not).

**Comment by sawenzel** [comment]
@mkrzewic : You are right. The constructors will have to be refined and we have to take care of ownership. This class should be understood as a demonstrator/prototype for further developments. I wanted to demonstrate ROOT IO and have something compiling ... first.

**Pull request merged**